x/auth/types/stdsignmsg.go

```diff
@@ -8,15 +8,16 @@ import (
 // a Msg with the other requirements for a StdSignDoc before
 // it is signed. For use in the CLI.
 type StdSignMsg struct {
```

Comment:
Note: This PR changes the `StdSignMsg`. Is there any way to avoid this or make it more general than just a `FeeAccount`? What other types of delegations?

Reply:
There is no other way to add this info without changing the sign bytes.
The info is tx global (not msg specific), and thus needs to be there.

We can also stop using and concrete structs and use only interfaces, and architect it in a way that we use reflection to detect, but that is a much deeper change. Eg. we never check StdSignMsg, StdTx, etc as arguments, just abstract interfaces.

And add a DelegatedSignMsg with this extra `FeeAccount` field and a `GetFeeAccount()` method. The AnteHandler can use reflection to see if the implementation has this method or not. And will allow various tx types coming in.

That said, this also seems dangerous to allow modules to register new tx types, with possilbly different sign byte calculations, etc. And would require a deep ADR approach.

I must say this approach was done months ago in #4616 and not flagged by anyone

Reply:
Looking more into the code, I could easily just modify `StdFee` to have one more field for the payer, which would probably be cleaner. I just followed the approach Aaron had done as I believed it had been discussed with the sdk team.

Modifying StdFee, even with an optional field, will change StdTx, however it will still accept the old format without any changes, and this will be a lot less breaking to many of the interfaces. If you prefer that, I can update the code.

I think that is also much less invasive than try to abstract out the field in some optional interface and have multiple different Tx types that are switched between at runtime.